qutip/core/qobj.py

```diff
@@ -391,7 +394,9 @@ def to(self, data_type):
             converter = _data.to[data_type]
         except (KeyError, TypeError):
             raise ValueError("Unknown conversion type: " + str(data_type))
-        if type(self._data) is data_type:
+        if type(self._data) is data_type and copy:
```

Comment:
While you say this, I just realized `data_type` could be an alias instead of a type, but it's not documented and the check for no change is incomplete...
This feels outside the PR but I will fix it while we are at it.